staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go

```diff
@@ -94,12 +95,26 @@ func UpdateResource(r rest.Updater, scope RequestScope, admit admission.Interfac
 		}
 
 		userInfo, _ := request.UserFrom(ctx)
-		staticAdmissionAttributes := admission.NewAttributesRecord(nil, nil, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo)
 		var transformers []rest.TransformFunc
-		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok && mutatingAdmission.Handles(admission.Update) {
+		if mutatingAdmission, ok := admit.(admission.MutationInterface); ok {
 			transformers = append(transformers, func(ctx context.Context, newObj, oldObj runtime.Object) (runtime.Object, error) {
-				return newObj, mutatingAdmission.Admit(admission.NewAttributesRecord(newObj, oldObj, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+				if oldObj == nil {
+					return nil, fmt.Errorf("unexpected nil object in mutating admission check of UPDATE request")
+				}
+				accessor, err := meta.Accessor(oldObj)
```

Comment:
I might be totally wrong here :) but if the object is being "created" here, won't `oldObj` be always nil?